src/controller/CHIPDeviceController.cpp

```diff
@@ -151,6 +151,22 @@ CHIP_ERROR DeviceController::Init(ControllerInitParams params)
 
     ReturnErrorOnFailure(ProcessControllerNOCChain(params));
 
+    DeviceProxyInitParams deviceInitParams = {
+        .sessionManager = params.systemState->SessionMgr(),
+        .exchangeMgr    = params.systemState->ExchangeMgr(),
+        .idAllocator    = &mIDAllocator,
+        .fabricInfo     = params.systemState->Fabrics()->FindFabricWithIndex(mFabricIndex),
+        .imDelegate     = params.systemState->IMDelegate(),
+    };
+
+    CASESessionManagerInitParams sessionManagerParams = {
+        .sessionInitParams = deviceInitParams,
+        .dnsCache          = &mDNSCache,
+    };
+
+    mCASESessionManager = chip::Platform::New<CASESessionManager>(sessionManagerParams);
+    VerifyOrReturnError(mCASESessionManager != nullptr, CHIP_ERROR_NO_MEMORY);
```

Comment:
Since this will be used by embedded devices, we are now managing sessions with heap here. There is not global tracking of the number of case sessions to be able to meet the "minimum to support" limits called by the spec (3 CASE sessions per fabric in/out). I actually am not sure how we will make sure that at least 3 are always available per fabric. The intent was that existing sessions be dropped (LRU-basis) if a new session is needed and contexts are not available. It would be good to understand if running out of heap here could have a side-effect of infrequent but extremely hard to debug failure of reachability... Since this feature will be used to ensure CASE session establishments to OTA providers, it might be an obstacle to reliable OTA in some situations.

Reply:
Added a TODO to CASESessionManager to add LRU based session eviction.
This API is instantiated at the start of the application/device. And, it is not reinitialized during run time. So memory allocation should be more deterministic than a typical runtime heap allocation.